arbor/threading/threading.hpp

```diff
@@ -161,7 +162,8 @@ class ARB_ARBOR_API task_system {
     task_system();
 
     // Create nthreads-1 new std::threads running run_tasks_loop(tid)
-    task_system(int nthreads);
+    task_system(int nthreads, bool bind_threads=false);
```

Comment:
I wonder weather the bind option should be `true` by default? Is there a common use case where this may not be the best option?

Reply:
I had an internal debate about precisely this. I went with off by default since most users likely don't 
care on their laptop w/ 2 cores, this would change a default, and might raise an error.